acquia/acquia_cms
# ACMS-707: Filter type are not getting filtered properly when filtered via Facets

**Motivation**

_Facet Pretty Path Default working:_
Facet pretty path by default adds tid to the term while creating its alias. For e.g. `Meet Up` term alias will be `meet-123` where 123 is the term id. While decoding the alias into term, it takes the tid and use it for filtering purpose. 

_ACMS Customisation_
In ACMS, for the clean URLs, the default beahviour is further modified to remove the `tid` from the alias and do a database query to obtain the term id. So, according to Acquia CMS `Meet Up` term alias will be `meet` and then a database query is used to obtain the term which should be an exact match of meet.

_Issue_
Since ACMS removes the tid from the alias, and database query is an exact match query. `Meet` never matches to `Meet Up` and hence no results are obtained.

Fixes #NNN

**Proposed changes**


- Remove custom facet pretty path plugin
- Use the default coder provided by facet pretty path module.

**Alternatives considered**

- Do not use pathauto's cleanstring, rather create own cleanString method to clean the terms for pretty paths. 
- Instead of exact database search, to a pattern match. Process the terms return by pattern by using cleanString method and then return the term id.


**Testing steps**


- Go to events view and click on ,meet-up filter, and check if results are being fetched
- Verify all view pages and check each term provided in facet pretty path block and make sure all are working as exptect.

**Merge requirements**
- [_Bug_] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
